students/HABTAMU/session07/html_render.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+
+"""
+A class-based system for rendering html.
+"""
+
+# This is the framework for the base class
+class Element(object):
+
+    tag = "html"
+    indent = "  "
+    # html_attributes = {"pid": "intro", "style": "text-align: center", "width":"400"}
+    html_attributes = dict()
+
+    def __init__(self, content=None, **kwargs):
+        self.html_attributes = kwargs
+
+        if content is None:
+            self.contents = []
+        else:
+            self.contents = [content]
+        # print("contents is:", self.contents)
+
+
+    def append(self, new_content):
+        self.contents.append(new_content)
+        
+    def render(self, out_file, ind=""):
+        # loop through the list of contents
+        attribute_str = "".join([f' {attribute}="{value}"' for attribute,value in self.html_attributes.items()])
+        out_file.write(ind + "<{}{}>\n".format(self.tag,attribute_str))
+        for content in self.contents:
+            # Duck Typing on the Fly
+            try:
+                content.render(out_file, ind + "  ")
```

Comment:
don't hard-code that indentation! 

It should be a class attribute on the Element class